src/ladds/Simulation.cpp

```diff
@@ -198,10 +208,46 @@ std::tuple<CollisionFunctor::CollisionCollectionT, bool> Simulation::collisionDe
   return {collisionFunctor.getCollisions(), stillTuning};
 }
 
+CollisionFunctor::CollisionCollectionT Simulation::collisionDetectionImmigrants(
+    AutoPas_t &autopas,
+    std::vector<Particle> &incomingParticles,
+    double deltaT,
+    double maxV,
+    double collisionDistanceFactor,
+    double minDetectionRadius) {
+  using autopas::utils::ArrayMath::abs;
+  using autopas::utils::ArrayMath::add;
+  using autopas::utils::ArrayMath::div;
+  using autopas::utils::ArrayMath::mul;
+  using autopas::utils::ArrayMath::mulScalar;
+  using autopas::utils::ArrayMath::sub;
+
+  CollisionFunctor collisionFunctor(autopas.getCutoff(), deltaT, collisionDistanceFactor, minDetectionRadius);
+  const std::array<double, 3> maxVVec{maxV, maxV, maxV};
+  const auto maxCoveredDistance = mulScalar(maxVVec, deltaT);
+
+  for (auto &immigrant : incomingParticles) {
+    // find the bounding box around the path the immigrant took since the last update
+    // pos - velocity * time
+    const auto posFirstPotentialCollision = sub(immigrant.getPosition(), mulScalar(immigrant.getVelocity(), deltaT));
+    // pos + maxVelocity * time
+    const auto higherCorner = add(posFirstPotentialCollision, maxCoveredDistance);
+    // pos - maxVelocity * time
+    const auto lowerCorner = sub(posFirstPotentialCollision, maxCoveredDistance);
+
+    // interact the immigrant with everything in the box
+    autopas.forEachInRegion(
```

Comment:
Depending how (in)efficient this is we could find a smarter, lower estimate based on the particles current velocity / energy